core/src/main/kotlin/pages/contentNodeProperties.kt

```diff
@@ -7,10 +7,40 @@ class SimpleAttr(val extraKey: String, val extraValue: String) : ExtraProperty<C
     override val key: ExtraProperty.Key<ContentNode, SimpleAttr> = SimpleAttrKey(extraKey)
 
     companion object {
-        fun header(value: String) = SimpleAttr("data-togglable", value)
+        fun togglableTarget(value: String) = SimpleAttr(TOGGLEABLE_CONTENT_TYPE_ATTR, value)
     }
 }
 
+object HtmlInvisibleExtra : ExtraProperty<ContentNode>, ExtraProperty.Key<ContentNode, HtmlInvisibleExtra> {
+    override val key: ExtraProperty.Key<ContentNode, *> = this
+}
```

Comment:
I do not like the CSS approach.
- It is impossible to understand from the code hat some headers are invisible in HTML.  Also, it violates our rendering model.
- The style  `.main-content[content-type="package"] .tabs-section-body > h2` can be decayed as well.



 Maybe `HtmlInvisibleExtra` is a bad name.
 Since the visibility of headers is related to the support of tabs in an output format, we can introduce a specific `TabbedContentType`. WDYT?

Reply:
> It is impossible to understand from the code hat some headers are invisible in HTML

I'm afraid, that's how all modern frontend works - you are not able to understand from code what the page will look like, because styles. We can apply an `invisible` style to headers if it's only in the HTML format itself - I don't mind that, as long as it doesn't leak in the IR layer. The proposal with `.main-content[content-type="package"]` )or similar) to me looks more reusable and universal though.

> Also, it violates our rendering model.

How so?

> The style .main-content[content-type="package"] .tabs-section-body > h2 can be decayed as well.

It indeed can be, that's why we need to have UI tests. But this will be a local problem of the HTML format that should be easier to spot, rather than a problem of all formats at once.

> Since the visibility of headers is related to the support of tabbed content in an output format, so a specific ToggleableContentType can be introduced (e.g. INVISIBLE).

I see what you mean by suggesting `ToggleableContentType.INVISIBLE`, it will work for the HTML format because of how `.js` scripts work for toggling content/tabs: "show only X", and `INVISIBLE` will not be part of X, so it's not rendered.

However, imagine I'm writing a `Json` output format, and I see `ToggleableContentType.INVISIBLE` - what should I do with it? How should I handle it?

Reply:
> I'm afraid, that's how all modern frontend works - you are not able to understand from code what the page will look like, because styles. 

It is not only about styles. It is about a structure, DOM etc.
Do you expect that all (or only the first one) headers are hidden here?
```
tabContent{
header{ "Header 1 " }
// some content
header{ "Header 2 " }
// some content
}

```


> How so?

We render something that should not be displayed. The model does not reflect the actual representation, 

> I see what you mean by suggesting ToggleableContentType.INVISIBLE, it will work for the HTML format because of how .js scripts work for toggling content/tabs: "show only X", and INVISIBLE will not be part of X, so it's not rendered.

Yes, the possible solution. Or do not render everything with `ToggleableContentType.INVISIBLE` in HTML render at all.

> However, imagine I'm writing a Json output format, and I see ToggleableContentType.INVISIBLE - what should I do with it? How should I handle it?

I do not know the implementation a tabbed content in JSON. Anyway, it can be ignored like in GFM.

Reply:
> We render something that should not be displayed.

That's a valid point, although I think it's fine if applied conditionally like in our case. 

If you don't want to render headers at all, I think it can be done with overriding `buildHeader` in the `HtmlRenderer`, the function that has the `context` param. Based on that context, the header rendering can be skipped. It should be essentially the same thing as the `isClasslike` boolean param.

Or maybe we don't have to add headers in the intermediate model at all, just skip them. 

In the HTML/GFM renderers, we can add a `node.extra[ToggleableContentTypeExtra]` case to `wrapGroup`, and then add an artificial header for the whole section. I like this even better, tbh.

Something like:

```kotlin
node.extra[ToggleableContentTypeExtra] != null -> div {
    val shouldRenderHeader = pageContext is ClasslikePage
    if (shouldRenderHeader) {
        val headerTitle = node.extra[ToggleableContentTypeExtra].value.toHeader()
        h2 { text(headerTitle) }
        childrenCallback()
    }
}
```

I'm not arguing that CSS is the only way to go, but what we want visually should be possible to implement without `isClasslike`, `isVisibleHeader` and similar details leaking into the common layer, as at the moment it's an implementation detail of the HTML format only.

Down the road, deprecating these parameters and `HtmlInvisible`-like extras will be significantly more difficult than dealing with headers hidden by CSS, as few people depend on it and we don't (yet) give any guarantee for the DOM structure.

Reply:
```
node.extra[ToggleableContentTypeExtra] != null -> div {
    val shouldRenderHeader = pageContext is ClasslikePage
    if (shouldRenderHeader) {
        val headerTitle = node.extra[ToggleableContentTypeExtra].value.toHeader()
        h2 { text(headerTitle) }
        childrenCallback()
    }
}
```
This will not work since `childrenCallback` renders all tabs at once. 
It can require changing a rendering logic to filter each tab in `children`. Is it worth it?

Reply:
Same as with CSS, there might be other ways, that's just a proposal/idea.

But I think it is worth it to get rid of `isClasslike`, `isVisibleHeader` and `HtmlInvisibleExtra` in public/intermediate API - that's my primary concern